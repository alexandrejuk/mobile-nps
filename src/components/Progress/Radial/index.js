import React from 'react'
import PropTypes from 'prop-types'

import styles from './style.module.css'

const calcProgress = value => (
  value > 0
    ? (440 - 50 * (value / 100))
    : 440
)

const Radial = ({
  cy,
  cx,
  r,
  progress,
}) => (
  <div className={styles.percent}>
    <svg>
      <circle cx={cx} cy={cy} r={r} />
      <circle
        data-value={progress}
        style={{
          strokeDashoffset: calcProgress(progress),
        }}
        cx={cx}
        cy={cy}
        r={r}
      />
    </svg>
  </div>
)

Radial.propTypes = {
  cx: PropTypes.number.isRequired,
  cy: PropTypes.number.isRequired,
  r: PropTypes.number.isRequired,
  progress: PropTypes.number.isRequired,
}

export default Radial
