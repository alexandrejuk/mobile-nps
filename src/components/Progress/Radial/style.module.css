@import '../../../assets/css/colors.css';

.percent {
  position: relative;
  width: 30px;
  height: 30px;
}

.percent svg {
  position: relative;
  width: 30px;
  height: 30px;
}

.percent svg circle {
  width: 30px;
  height: 30px;
  fill: none;
  stroke-width: 2;
  stroke: #000;
  transform: translate(5px, 5px);
  stroke-dasharray: 440;
  stroke-dashoffset: 440;
  stroke-linecap: round;
}

.percent svg circle:nth-child(1) {
  stroke-dashoffset: 0;
  stroke: var(--color-concrete);
}

.percent svg circle:nth-child(2) {
  stroke: var(--color-cardinal);
}
