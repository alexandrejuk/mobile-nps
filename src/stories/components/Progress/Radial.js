import React from 'react'
import { Progress } from '../../../components'
import styles from './style.module.css'

export default {
  title: 'Components/Progress/Radial',
}

export const Default = () => (
  <div className={styles.container}>
    <div className={styles.radial}>
      <Progress
        progress={25}
        setup={{
          type: 'radial',
          radial: {
            cx: 8,
            cy: 8,
            r: 8,
          },
        }}
      />
      <h4>25 %</h4>
    </div>
    <div className={styles.radial}>
      <Progress
        progress={50}
        setup={{
          type: 'radial',
          radial: {
            cx: 8,
            cy: 8,
            r: 8,
          },
        }}
      />
      <h4>50 %</h4>
    </div>
    <div className={styles.radial}>
      <Progress
        progress={75}
        setup={{
          type: 'radial',
          radial: {
            cx: 8,
            cy: 8,
            r: 8,
          },
        }}
      />
      <h4>75 %</h4>
    </div>
    <div className={styles.radial}>
      <Progress
        progress={100}
        setup={{
          type: 'radial',
          radial: {
            cx: 8,
            cy: 8,
            r: 8,
          },
        }}
      />
      <h4>100 %</h4>
    </div>
  </div>
)
