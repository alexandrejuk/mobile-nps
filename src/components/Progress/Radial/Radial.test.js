import React from 'react'
import { render } from '@testing-library/react'

import Radial from './'

describe('Radial component', () => {
  it('should render correctly', () => {
    const { container } = render(
      <Radial
        progress={25}
        cx={8}
        cy={8}
        r={8}
      />
    )
    expect(container).toMatchSnapshot()
  })
})
