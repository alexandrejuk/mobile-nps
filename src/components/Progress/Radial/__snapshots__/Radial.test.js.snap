// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radial component should render correctly 1`] = `
<div>
  <div
    class="percent"
  >
    <svg>
      <circle
        cx="8"
        cy="8"
        r="8"
      />
      <circle
        cx="8"
        cy="8"
        data-value="25"
        r="8"
        style="stroke-dashoffset: 427.5;"
      />
    </svg>
  </div>
</div>
`;
